File: Python/BOJ/스택의활용/10799.py
arr = list(input())

count = 0; result = 0
for i in range(len(arr)):
  if arr[i] == "(":
    count += 1
  else:
    prev = arr[i-1]
    count -= 1
    if prev == "(":
      result += count
    else:
      result += 1

print(result)

File: Swift/BOJ/문제집/스택/10799.swift
//
//  10799.swift
//  Algorithm
//
//  Created by 유정주 on 2023/07/08.
//

import Foundation

let arr = Array(readLine()!)
var count = 0, result = 0
for i in 0..<arr.count {
    if arr[i] == "(" {
        count += 1
    } else {
        count -= 1
        if arr[i-1] == "(" {
            //레이저일 때
            result += count
        } else {
            //파이프 끝일 때
            result += 1
        }
    }
}

print(result)
